dataclients/kubernetes/kube.go

```diff
@@ -383,6 +384,25 @@ func ParsePathMode(s string) (PathMode, error) {
 	}
 }
 
+// ParseTLSClientAuth parses the string representations of different
+// client auth types.
+func ParseTLSClientAuth(s string) (tls.ClientAuthType, error) {
+	switch s {
```

Comment:
```suggestion
	switch strings.ToLower(s) {
```
just an idea to support casing as well